### DISCUSSION TITLE
Need to access Assistant Context from Task or a Script

### DISCUSSION THREAD

**Discussion by Cheirnomus** [zed-industries/zed#24680] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

Is it possible to use a Task variable to access the Assistant Context so a script run by the same Task can then find and extract specific tagged chunks of data generated by the LLM?

How can I address / connect to a tab / file in the Assistant from a Task?

### Environment

Mac M3 Sonoma 14.6.1 and Zed v0.162.5

**Comment by notpeter** [comment]
Are you looking for the complete context as a JSON object stuffed into an environment variable?  On macOS that would be limited to ~1MB (`getconf -a | grep ARG_MAX`).

Can you expand a little more about your use case?

**Comment by Cheirnomus** [comment]
Sure, here is my use case. I'm generating apps using a custom prompt template and the output code blocks can belong to different project file types. The blocks are tagged according to their files extension type, so they are ready to find and extract.

For now, I have to copy and paste all the output that the template and LLM generated to a MD file. This manual step is required because my script can not access the Assistant Panel context (I only need access to the generated code not necessarily the whole Assistant history file).

Once the copy is made I run my project_builder script to automatically generate the project layout and put the file type sections into the right directories in the project tree.

I'm hoping to generate the code using the template in the Assistant Panel then go straight to the terminal to run my project_builder script. This way the project is more than stubbed out plus built and ready to run and develop further using the Inline Assistant and Supermaven.

Using a Task to setup the project and run the project_builder script is the ultimate goal, instead of even having to involve the terminal.

**Comment by Cheirnomus** [comment]
Any news on a fix for getting text out of the Assistant context using a Task and/or script?

There was a interesting relevant mention about copying and pasting I recently saw in the latest Zed blog post.

"was to copy/paste like we did in the Dark Ages"

Blog link: https://zed.dev/blog/mcp
